python/lsst/analysis/tools/interfaces/_stages.py

```diff
@@ -218,12 +218,19 @@ def __call__(self, data: KeyedData, **kwargs) -> MetricResultType:
             if formattedKey not in data:
                 raise ValueError(f"Key: {formattedKey} could not be found input data")
             value = data[formattedKey]
-            if not isinstance(value, Scalar):
-                raise ValueError(f"Data for key {key} is not a Scalar type")
```

Comment:
Curious why you removed this `raise`, rather than add `Mapping` as an exception and keep it?

Reply:
The `raise` below will only trigger now if `isinstance(value, Mapping)` is `True`, which doesn't capture the same functionality as this prior `raise`.

Reply:
I see it in the `else` block now - I suppose that achieves the same thing, but I did prefer failing sooner rather than later.

Reply:
Correct, I just moved it elsewhere. Also, my tests show that `isinstance(value, Scalar)` doesn't work as expected:
```python
>>> from lsst.analysis.tools.interfaces import Scalar
>>> isinstance("string", Scalar)
True
>>> isinstance(["string1","string2"], Scalar)
True
>>> isinstance([1, 2, 3], Scalar)
True
>>> isinstance(5.5, Scalar)
True
>>> isinstance([5.5, 8.7], Scalar)
True
>>> isinstance({}, Scalar)
True
>>> isinstance({"key1": 2.60, "key2": 9.13}, Scalar)
True
```

Reply:
@natelust do you expect this? The code for `Scalar` seems to imply that it's meant to only be numbers and not any other type.

Reply:
@natelust is on vacation at present, and is unlikely to respond before this ticket is merged. My suggestion is for this to be flagged on a new ticket, and addressed once @natelust gets back.

Reply:
I just showed this to @sr525, and she thinks we should file a ticket for it and have @natelust  look at it. I'll file one.

Reply:
Filed [DM-45309](https://rubinobs.atlassian.net/browse/DM-45309) and added @natelust as a watcher.